test/api_integration/apis/console/proxy_route.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+
+  describe('POST /api/console/proxy', () => {
+    describe('system indices behavior', () => {
+      it('returns warning header when making requests to .kibana index', async () => {
+        return await supertest
+          .post('/api/console/proxy?method=GET&path=/.kibana/_settings')
+          .set('kbn-xsrf', 'true')
+          .then((response) => {
+            expect(response.header).to.have.property('warning');
+            const { warning } = response.header as { warning: string };
+            expect(warning.startsWith('299')).to.be(true);
+            expect(warning.includes('system indices')).to.be(true);
+          });
+      });
+
+      it('does not forward x-elastic-product-origin', async () => {
+        // If we pass the header and we still get the warning back, we assume that the header was not forwarded.
```

Comment:
This is a bit fragile but seemed like the only practical option that didn't expose a big security hole or add testing-specific code to the implementation. This test isn't going to live for very long, so I feel think it's fine?

Reply:
Is there a test in Core proofing we don't receive a warning when requesting ES with `x-elastic-product-origin` header?

Reply:
We didn't have an integration test so I have added one that verifies both with and without the header to make sure everything is consistent.